docs/redis.md

```diff
@@ -1,5 +1,94 @@
+- `model_id`: a unique ID for the Cog model, used to label setup logs.
+- `log_queue`: the queue the Cog model should send setup logs to (prediction logs are sent as part of prediction responses).
+
+Note: logging is changing as part of 0.3.0, so the `model_id` and `log_queue` arguments are likely to change soon.
+
+You can optionally provide the following positional argument:
+
+- `predict_timeout`: the maximum time in seconds a prediction will be allowed to run for before it is terminated.
+
+For example:
+
+    docker run python -m cog.server.redis_queue \
+        redis 6379 predict-queue \
+        https://example.com/ab48b7ff-1589-4360-a54b-47f9d8d3f6b7/ \
+        worker-1 \
+        widget-classifier logs-queue \
+        120
+
+After starting, [the `setup()` method of the predictor](python.md#predictorsetup) is called. When setup is finished the model will start polling the input queue for prediction request messages.
+
+## Enqueue a prediction
+
+The message body should be a JSON object with the following fields:
+
+- `inputs`: a JSON object with the same keys as the [arguments to the `predict()` function](python.md). Any `File` or `Path` inputs are passed as URLs.
+- `id`: ...
+- `response_queue`: ...
+
+TODO: rename `inputs` to `input` to match the HTTP API.
+
+TODO: example call to Redis using XADD
+
+## Get a prediction response
+
+The model will send a message to the queue every time something happens:
+
+- when the model generates some logs
+- when the model returns some output
+- when the model finishes running
+
+TODO: example call to Redis using RPOP (or equiv)
+
+The message body is a JSON object with the following fields:
+
+- `status`: `processing`, `succeeded` or `failed`.
+- `output`: The return value of the `predict()` function.
+- `logs`: A list of any logs sent to stdout or stderr during the prediction.
+- `error`: If `status` is `failed`, the error message.
+
+For example, a message early in the prediction might look like:
+
+    {
+        "status": "processing",
+        "output": null,
+        "logs": [
+            "Creating model and diffusion.",
+            "Done."
+        ]
+    }
+
+If the model yields [progressive output](python.md#progressive-output) then a mid-prediction message might look like:
+
+    {
+        "status": "processing",
+        "output": [
+            "https://example.com/ab48b7ff-1589-4360-a54b-47f9d8d3f6b7/0.jpg",
+            "https://example.com/ab48b7ff-1589-4360-a54b-47f9d8d3f6b7/20.jpg",
+            "https://example.com/ab48b7ff-1589-4360-a54b-47f9d8d3f6b7/40.jpg",
+        ],
+        "logs": [
+            "Creating model and diffusion.",
+            "Done.",
+            "Iteration: 0, loss: -0.767578125",
+            "Iteration: 20, loss: -1.2333984375",
+            "Iteration: 40, loss: -1.380859375"
+        ]
+    }
+
+TODO: document `setup_time` and `run_time` queues?
```

Comment:
I think we're going to get rid of this in favor of putting metrics in the response? May as well leave undocumented.